be1-go/hub/rollCall.go

```diff
@@ -5,29 +5,28 @@ import (
 	"log"
 	"os"
 	"student20_pop/message"
+	"student20_pop/message2/messagedata"
+	messageX "student20_pop/message2/query/method/message"
 
 	_ "github.com/mattn/go-sqlite3"
 	"golang.org/x/xerrors"
 )
 
 // processCreateRollCall processes a roll call creation object.
-func (c *laoChannel) processCreateRollCall(data message.Data) error {
-
-	rollCallData := data.(*message.CreateRollCallData)
-
+func (c *laoChannel) processCreateRollCall(msg messagedata.RollCallCreate) error {
 	// Check that the ProposedEnd is greater than the ProposedStart
-	if rollCallData.ProposedStart > rollCallData.ProposedEnd {
-		return message.NewErrorf(-4, "The field `proposed_start` is greater than the field `proposed_end`: %d > %d", rollCallData.ProposedStart, rollCallData.ProposedEnd)
+	if msg.ProposedStart > msg.ProposedEnd {
+		return message.NewErrorf(-4, "The field `proposed_start` is greater than the field `proposed_end`: %d > %d", msg.ProposedStart, msg.ProposedEnd)
 	}
 
-	c.rollCall.id = string(rollCallData.ID)
+	c.rollCall.id = string(msg.ID)
 	c.rollCall.state = Created
 	return nil
 }
 
 // processOpenRollCall processes an open roll call object.
-func (c *laoChannel) processOpenRollCall(data message.Data, action message.RollCallAction) error {
-	if action == message.RollCallAction(message.OpenRollCallAction) {
+func (c *laoChannel) processOpenRollCall(msg messageX.Message, action string) error {
+	if action == "open" {
```

Comment:
What do you think about having a big string enum somewhere with the different options ? 
It seems a bit brittle to have the strings hardcoded in possibly multiple places.

Reply:
Yes, there is definitely something to do about that. I'm waiting to see how things come up before entering the "code polishing phase" 😄.